src/ncp/ncp_base.cpp

```diff
@@ -414,8 +416,8 @@ void NcpBase::HandleSendDone()
     if (mSendQueue.GetHead() != NULL)
     {
         Message &message(*mSendQueue.GetHead());
-        HandleDatagramFromStack(message);
         mSendQueue.Dequeue(message);
+        HandleDatagramFromStack(message);
```

Comment:
Yeah, but to we want to call it for datagrams which the stack itself emitted?

In any case, this particular discussion shouldn't block the rest of this commit, so I'll go ahead and approve this.